### ISSUE TITLE
Name in notifications is different to name signed up to event

### ISSUE THREAD

**Issue by moehike** [WarwickTabletop/tgrsite#274]
I put up the "Trashkin 2021" event. I got a notification for user Santiella signing up for the event. However, when I click on the event, the user that has signed up is Kiwi_and_Lime. There are no notifications for anyone de-signing up. There are 3 other people signed up to the event, all of whose name is consistent across the notification and the event. I have attached screenshots of the notifications and the event page.
![image](https://user-images.githubusercontent.com/59832236/128647648-16c19333-67cb-4435-a2c4-97e3830dcf80.png)
![image](https://user-images.githubusercontent.com/59832236/128647660-aa4de363-44b1-4b2b-b407-f3076f3c9097.png)

**Comment by finnbar** [comment]
Thank you for your report!

I think I can guess the source of this - the notification is just text created at the time of the event, so if a user changes their name (which I'm guessing is what happened here), the notification isn't updated as it has no link to the user. An overhaul of notifications would be needed to include the link to the user rather than just storing it as text in the database, so this is something for the far future.

**Comment by the-Bruce** [comment]
Yeah, to fix this you'd probably have to either come up with a very clever data model, or just give up on 1NF and store the data as JSON in the table to be expanded by the server each view. The latter option is the most flexible, but does come at the cost of more complex rendering.
